anitya_schema/anitya_schema/project_messages.py

```diff
@@ -0,0 +1,544 @@
+                },
+                "required": [
+                    "backend",
+                    "created_on",
+                    "homepage",
+                    "id",
+                    "name",
+                    "regex",
+                    "updated_on",
+                    "version",
+                    "version_url",
+                    "versions",
+                ],
+            },
+        },
+    }
+
+
+class ProjectMapDeleted(message.Message):
+    topic = "org.release-monitoring.prod.anitya.project.map.remove"
+    body_schema = {
+        "id": "https://fedoraproject.org/jsonschema/anitya.project.map.deleted.json",
+        "$schema": "http://json-schema.org/draft-04/schema#",
+        "type": "object",
+        "required": ["project", "message", "distro"],
+        "properties": {
+            "distro": {"type": "null"},
+            "message": {
+                "type": "object",
+                "properties": {
+                    "agent": {"type": "string"},
+                    "distro": {"type": "string"},
+                    "project": {"type": "string"},
+                },
+                "required": ["agent", "distro", "project"],
+            },
+            "project": {
+                "type": "object",
+                "properties": {
+                    "backend": {"type": "string"},
+                    "created_on": {"type": "number"},
+                    "homepage": {"type": "string"},
+                    "id": {"type": "integer"},
+                    "name": {"type": "string"},
+                    "regex": {"anyOf": [{"type": "string"}, {"type": "null"}]},
+                    "updated_on": {"type": "number"},
+                    "version": {"anyOf": [{"type": "string"}, {"type": "null"}]},
+                    "version_url": {"anyOf": [{"type": "string"}, {"type": "null"}]},
+                    "versions": {"type": "array"},
+                },
+                "required": [
+                    "backend",
+                    "created_on",
+                    "homepage",
+                    "id",
+                    "name",
+                    "regex",
+                    "updated_on",
+                    "version",
+                    "version_url",
+                    "versions",
+                ],
+            },
+        },
+    }
+
+
+class ProjectVersionUpdate(message.Message):
+    topic = "org.release-monitoring.prod.anitya.project.version.update"
+    body_schema = {
+        "id": "https://fedoraproject.org/jsonschema/anitya.project.version.update.json",
+        "$schema": "http://json-schema.org/draft-04/schema#",
+        "type": "object",
+        "required": ["project", "message", "distro"],
+        "properties": {
+            "distro": {"type": "null"},
+            "message": {
+                "type": "object",
+                "properties": {
+                    "agent": {"type": "string"},
+                    "odd_change": {"type": "boolean"},
```

Comment:
This is no longer happening. I removed this error when I introduced check for multiple versions.

Reply:
That's not good :(

The message schema changed and it's possible consumers were relying on that field. If, for example, the-new-hotness expects this key to be present it will crash. The best path forward is likely to make sure that key remains the message getting sent out and is always false or something.

Reply:
Oh, I didn't knew this could have impact like this. The change is only on master right now. It wasn't released yet, so I can add the field back.
It's hard to know what is required.

Reply:
The odd_change is now always false.